cli/src/main.rs

```diff
@@ -198,6 +218,9 @@ fn get_file_paths(input_patterns: Vec<String>) -> Result<Vec<PathBuf>, GlobError
             };
             glob(&glob_pattern)
                 .expect("failed to read glob pattern")
+                .filter(|is_file| {
+                    is_file.as_ref().is_ok_and(|file| !exclude_patterns.iter().any(|pattern| pattern.matches_path(file)))
+                })
                 .collect::<Vec<_>>()
```

Comment:
not part of PR changes, but I think we can prevent allocation by not collecting to a vec here. `flat_map` takes a function that returns an `impl IntoIterator`, so returning the iterator instead should be fine.